File: examples_android/app/src/main/java/com/confirmit/testsdkapp/layouts/simpleCardLayout/items/SimpleCardSingleItem.kt
package com.confirmit.testsdkapp.layouts.simplecardlayout.items

import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import com.confirmit.mobilesdk.ui.questions.QuestionAnswer
import com.confirmit.mobilesdk.ui.questions.SingleQuestion
import com.confirmit.testsdkapp.R

class SimpleCardSingleItem(view: View) : SimpleCardItem(view), SimpleCardSelectedItemClickedListener {

    private lateinit var question: SingleQuestion

    private val listView: LinearLayout = view.findViewById(R.id.ctnList)
    private val itemList: MutableList<SimpleCardSelectedItem> = mutableListOf()

    fun setupQuestion(question: SingleQuestion) {
        this.question = question
        txtHeader.text = question.text.get()

        createAnswers(question)

        handleValidation(question.errors)
    }

    private fun createAnswers(question: SingleQuestion) {
        listView.removeAllViews()
        itemList.clear()
        val selectedAnswer = question.selected()
        question.answers.forEach { questionAnswer ->
            val view = LayoutInflater.from(getView().context).inflate(R.layout.simple_card_select_item, listView, false)
            val selectItem = SimpleCardSelectedItem(view)
            selectItem.setupItem(false, questionAnswer, questionAnswer.code == selectedAnswer?.code)
            selectItem.listener = this
            itemList.add(selectItem)
            listView.addView(view)
        }
    }

    override fun onNext(): Boolean {
        return validate(question)
    }

    override fun onBack(): Boolean {
        return true
    }

    override fun onRadioClicked(answer: QuestionAnswer) {
        question.select(answer)
        for (item in itemList) {
            item.updateSelected(answer.code == item.answer.code)
        }
    }

    override fun onCheckClicked(answer: QuestionAnswer, selected: Boolean) {
    }
}

File: examples_ios/MyApp/Sources/SurveyLayouts/SimpleCardLayout/Items/SimpleCardSingleItem.swift
import Foundation
import MobileSdk
import UIKit

class SimpleCardSingleItem: SimpleCardItem {
    @IBOutlet var stackView: UIStackView!
    @IBOutlet var cstrStackHeight: NSLayoutConstraint!
    
    private var question: SingleQuestion!
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }
    
    func setupQuestion(question: SingleQuestion) {
        self.question = question
        lblHeader.text = question.text.get()
        
        createAnswers(question: question)
        
        handleValidation(errors: question.errors)
    }
    
    private func createAnswers(question: SingleQuestion) {
        let selectedAnswer = question.selected()
        for answer in question.answers {
            let selectItem: SimpleCardSelectItem = Utils.fromNib()
            selectItem.setupItem(checkbox: false, answer: answer, selected: answer.code == selectedAnswer?.code)
            selectItem.delegate = self
            stackView.addArrangedSubview(selectItem)
        }
        cstrStackHeight.constant = CGFloat(question.answers.count) * 36
    }
    
    override func onNext() -> Bool {
        return validate(question: question)
    }
    
    override func onBack() -> Bool {
        return true
    }
}

extension SimpleCardSingleItem: SimpleCardSelectItemDelegate {
    func onRadioClicked(answer: QuestionAnswer) {
        question.select(answer: answer)
        for subView in stackView.arrangedSubviews {
            if let selectItem = subView as? SimpleCardSelectItem {
                selectItem.updateSelected(selected: answer.code == selectItem.answer.code)
            }
        }
    }
    
    func onCheckClicked(answer: QuestionAnswer, selected: Bool) {
        // Nothing to do
    }
}
